web/models/class/class_helper.ex

```diff
@@ -5,9 +5,32 @@ defmodule CoursePlanner.ClassHelper do
   use CoursePlanner.Web, :model
 
   alias CoursePlanner.{Repo, Class, Notifier, Attendance, Notifier.Notification}
-  alias Ecto.DateTime
+  alias CoursePlanner.Terms.Term
+  alias Ecto.{Changeset, DateTime, Date}
   alias Ecto.Multi
 
+  def validate_for_holiday(%{valid?: true} = changeset) do
+    class_date = changeset |> Changeset.get_field(:date) |> Date.cast!
+    offered_course_id = changeset |> Changeset.get_field(:offered_course_id)
+
+    term = Repo.one(from t in Term,
+      join: oc in assoc(t, :offered_courses),
+      preload: [offered_courses: oc],
+      where: oc.id == ^offered_course_id)
+
+    class_on_holiday? =
+      Enum.find(term.holidays, fn(holiday) ->
+        Date.cast!(holiday.date) == class_date
+        end)
+
+    if class_on_holiday? do
+      add_error(changeset, :offered_course_id, "You cannot create a class on holiday")
```

Comment:
I think it would make more sense to add the error to the `date` field of the class. What do you think?